gatekeeper/client/client.go

```diff
@@ -0,0 +1,67 @@
+// Package client provides the Gatekeeper bootstrap API
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/manifoldco/torus-cli/gatekeeper/apitypes"
+	"github.com/manifoldco/torus-cli/registry"
+)
+
+const (
+	gatekeeperAPIVersion = "v0"
+	clientTimeout        = time.Minute
+)
+
+type clientRoundTripper struct {
+	registry.DefaultRequestDoer // TODO: Could be a generic RequestDoer?
+}
+
+// Client is the Gatekeeper bootstrapping client
+type Client struct {
+	rt *clientRoundTripper
+}
+
+// NewClient returns a new client to a Gatekeeper host that can bootstrap this machine
+func NewClient(host string) *Client {
+	if !strings.HasSuffix(host, "/") {
+		host += "/"
+	}
+	return &Client{
+		rt: &clientRoundTripper{
+			DefaultRequestDoer: registry.DefaultRequestDoer{
+				Client: &http.Client{
+					Timeout: clientTimeout,
+				},
+				Host: host,
+			},
+		},
+	}
+}
+
+// Bootstrap bootstraps the machine with Gatekeeper
+func (c *Client) Bootstrap(provider string, bootreq interface{}) (*apitypes.BootstrapResponse, error) {
+	path := fmt.Sprintf("%s/%s/%s", gatekeeperAPIVersion, "machine", provider)
+	req, err := c.rt.NewRequest("POST", path, nil, bootreq)
+	if err != nil {
+		return nil, err
+	}
+
+	var bootresp apitypes.BootstrapResponse
+	resp, err := c.rt.Do(context.Background(), req, &bootresp)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if bootresp.Error != "" {
+		return nil, errors.New(bootresp.Error)
+	}
```

Comment:
Will this ever happen? if the response code is not 2XX, the RequestDoer will attempt to populate apitypes.Error:
https://github.com/manifoldco/torus-cli/blob/master/registry/round_tripper.go#L107